mlflow/tracking/fluent.py

```diff
@@ -3129,6 +3164,13 @@ def predict(model_input):
         traces = mlflow.search_traces(model_id=mlflow.get_active_model_id(), return_type="list")
         assert len(traces) == 1
     """
+    active_model = _set_active_model(name=name, model_id=model_id)
+    active_model_ctx = _ACTIVE_MODEL_CONTEXT.get()
+    active_model_ctx.set_by_user = True
```

Comment:
Instead of mutating `active_model_ctx`, can we use `_ACTIVE_MODEL_CONTEXT.set` to set a new context?